src/refresh_tokens/refreshtokens.service.ts

```diff
@@ -0,0 +1,301 @@
+import { BadGatewayException, BadRequestException, Body, Header, Headers, Injectable, Logger, Options, Param } from "@nestjs/common";
+import { JwtService } from "@nestjs/jwt";
+import { PrismaService } from "src/prisma/prisma.service";
+import { refreshCookiesDTO, refreshDTO } from "./refreshToken.dto";
+import { userInfo } from "os";
+
+
+
+@Injectable()
+export class RefreshTokensService {
+    private readonly logger: Logger
+    constructor(
+        private readonly prismaService: PrismaService,
+        private readonly jwtService: JwtService
+    ) {
+        this.logger = new Logger();
+    }
+
+    async refreshToken(@Headers() cookie: refreshCookiesDTO, @Body() data: refreshDTO) {
+        if (!cookie || !data) {
+            return {
+                success: false,
+                message: 'please provide refresh and access token via cookie or body'
+            }
+        }
+        try {
+
+            const token = cookie.refreshToken ? cookie.refreshToken : data.refreshToken
```

Comment:
We can avoid a IO Db call here we can check here if there is token in any of those

Reply:
could not get it, please elaborate